[jit] Allow for segmented printing in PythonPrint

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #19239 [wip][jit] use fully qualified name for ScriptClasses
* **#19238 [jit] Allow for segmented printing in PythonPrint**
* #19237 [jit] add resolveType to Resolver
* #19236 [jit] Turn resolver into a class

Make printing to the `out` param a separate step, which allows us to
prepare several portions of the output separately and print them at the
end. There's only one for now (`body_`) but this is necessary for import
statements.

gh-metadata: pytorch pytorch 19238 gh/suo/26/head

Differential Revision: [D14928291](https://our.internmc.facebook.com/intern/diff/D14928291)

suo committed:
[jit] Allow for segmented printing in PythonPrint

Make printing to the `out` param a separate step, which allows us to
prepare several portions of the output separately and print them at the
end. There's only one for now (`body_`) but this is necessary for import
statements.

suo committed:
Update on "[jit] Allow for segmented printing in PythonPrint"

[jit] Allow for segmented printing in PythonPrint

Make printing to the `out` param a separate step, which allows us to
prepare several portions of the output separately and print them at the
end. There's only one for now (`body_`) but this is necessary for import
statements.

gh-metadata: pytorch pytorch 19238 gh/suo/26/head

suo committed:
Update on "[jit] Allow for segmented printing in PythonPrint"

[jit] Allow for segmented printing in PythonPrint

Make printing to the `out` param a separate step, which allows us to
prepare several portions of the output separately and print them at the
end. There's only one for now (`body_`) but this is necessary for import
statements.

gh-metadata: pytorch pytorch 19238 gh/suo/26/head

suo committed:
Update on "[jit] Allow for segmented printing in PythonPrint"

[jit] Allow for segmented printing in PythonPrint

Make printing to the `out` param a separate step, which allows us to
prepare several portions of the output separately and print them at the
end. There's only one for now (`body_`) but this is necessary for import
statements.

gh-metadata: pytorch pytorch 19238 gh/suo/26/head

suo committed:
Update on "[jit] Allow for segmented printing in PythonPrint"

[jit] Allow for segmented printing in PythonPrint

Make printing to the `out` param a separate step, which allows us to
prepare several portions of the output separately and print them at the
end. There's only one for now (`body_`) but this is necessary for import
statements.

gh-metadata: pytorch pytorch 19238 gh/suo/26/head

zdevito reviewed (approved):
Seems fine. Didn't look closely, but would be hard for it to work but be wrong.

suo committed:
Update on "[jit] Allow for segmented printing in PythonPrint"

[jit] Allow for segmented printing in PythonPrint

Make printing to the `out` param a separate step, which allows us to
prepare several portions of the output separately and print them at the
end. There's only one for now (`body_`) but this is necessary for import
statements.

gh-metadata: pytorch pytorch 19238 gh/suo/26/head
